### PR TITLE
Docs fixing and cleanup

### PR DESCRIPTION
**Pull request by PhilipVinc**
Removes 2 unused files and fixes a lot of `refs`-> `{class}/{meth}/{func}` or build errors in the docstrings

### CODE REVIEW

File: netket/utils/group/_group.py [hunk 1]
```diff
@@ -104,8 +104,15 @@ def remove_duplicates(self, *, return_inverse=False) -> "FiniteGroup":
...
-        :code:`gh = product(g, h)` is equivalent to :code:`Identity()`
+
+        Assuming the definitions
+
+        .. code::
+
+            g = self[idx_g]
+            h = self[self.inverse[idx_g]]
+
+        Then :code:`gh = product(g, h)` is equivalent to :code:`Identity()`
```
Comment (femtobit) [inline netket/utils/group/_group.py:115]:
```suggestion
        :code:`gh = product(g, h)` is equivalent to :code:`Identity()`
```

File: netket/utils/group/_group.py [hunk 2]
```diff
@@ -122,9 +129,16 @@ def inverse(self) -> Array:
...
+
+        Assuming the following definitions:
+
+        .. code::
+
+            g = self[idx_g]
+            h = self[idx_h]
+            idx_u = self.product_table[idx_g, idx_h]
+
+        then :code:`self[idx_u]` corresponds to :math:`u = g^{-1} h` .
```
Comment (femtobit) [inline netket/utils/group/_group.py:141]:
```suggestion
        Assuming the definitions

        .. code::

            g = self[idx_g]
            h = self[idx_h]
            idx_u = self.product_table[idx_g, idx_h]

        :code:`self[idx_u]` corresponds to :math:`u = g^{-1} h` .
```

File: netket/utils/group/_group.py [hunk 3]
```diff
@@ -141,8 +155,17 @@ def product_table(self) -> Array:
...
+
+        Assuming the definitions
+
+        .. code::
+
+            g = self[idx_g]
+            h = self[idx_h]
+
+        Then, :code:`self[self.conjugacy_table[idx_g,idx_h]]`
+        corresponds to :math:`h^{-1}gh`.
```
Comment (femtobit) [inline netket/utils/group/_group.py:168]:
```suggestion
        :code:`self[self.conjugacy_table[idx_g,idx_h]]`
        corresponds to :math:`h^{-1}gh`.
```

File: docs/docs/varstate.md [hunk 4]
```diff
@@ -14,10 +14,8 @@ are interested in implementing a new kind of variational state that encodes a st
 
 ## Constructing a Variational State
 
-To construct the two variational states above you need to provide at least a Monte Carlo sampler (you can see
-the list of available ones {ref}`here <sampler-api>`) and a model.
-The hilbert space of the variational state will be inferred from the sampler, as they all reference the
-hilbert space they are sampling.
+To construct the two variational states above you need to provide at least a Monte Carlo sampler (you can find the list of available ones [here](netket_sampler_api)) and a model.
+The hilbert space of the variational state will be inferred from the sampler, as they all reference the hilbert space they are sampling.
```
Comment (femtobit) [inline docs/docs/varstate.md:18]:
```suggestion
The Hilbert space of the variational state will be inferred from the sampler, which contains a reference to the underlying Hilbert space.
```

File: docs/docs/install.md [hunk 5]
```diff
@@ -208,29 +175,24 @@ The first thing to do is import NetKet. We usually shorten it to `nk`.
...
 hi = nk.hilbert.Spin(s=1 / 2, N=g.n_nodes)
 ha = nk.operator.Ising(hilbert=hi, graph=g, h=1.0)
 ```
 
-Then, one must define the hilbert space and the hamiltonian. Common options
-for the Hilbert spacee are {ref}`Spin`,  {ref}`Fock` or {ref}`QuBit`, but it is
-also possible to define your own. Those classes are contained in the {ref}`Hilbert`
-submodule.
+Then, one must define the hilbert space and the hamiltonian. 
+Common options for the Hilbert spacee are {class}`~nk.hilbert.Spin`,  {class}`~nk.hilbert.Fock` or {class}`~nk.hilbert.QuBit`, but it is also possible to define your own.
```
Comment (femtobit) [inline docs/docs/install.md:191]:
```suggestion
Common options for the Hilbert space are {class}`~nk.hilbert.Spin`,  {class}`~nk.hilbert.Fock` or {class}`~nk.hilbert.Qubit`, but it is also possible to define your own.
```

File: docs/docs/install.md [hunk 6]
```diff
@@ -208,29 +175,24 @@ The first thing to do is import NetKet. We usually shorten it to `nk`.
...
 
-Then, one must define the system to be studied. To do so, the first
-thing to do is usually defining the lattice of the model. This is not
-always required, but it can sometimes avoid errors.
-Several types of Lattices (graphs) are defined in the {ref}`Graph`
-submodule.
+Then, one must define the system to be studied.
+To do so, the first thing to do is usually defining the lattice of the model.
+This is not always required, but it can sometimes avoid errors.
+Several types of Lattices (graphs) are defined in the [Graph submodule](netket_graph_api).
```
Comment (femtobit) [inline docs/docs/install.md:181]:
```suggestion
For a discrete system, the first thing to do is usually defining the underlying lattice structure of the model.
Several types of lattices and, more generally, graphs allowing arbitrary connections between sites are defined in the [Graph submodule](netket_graph_api).
```

File: docs/docs/install.md [hunk 7]
```diff
@@ -154,46 +154,13 @@ to all relevant parts of the documentation.
 
 ## Jax/Flax extensions
 
-Netket v3 API is centered around [flax](https://flax.readthedocs.io), a jax library to simplify the definition and
-usage of Neural-Networks.
+Netket v3 API is centered around [flax](https://flax.readthedocs.io), a jax library to simplify the definition and usage of Neural-Networks.
```
Comment (femtobit) [inline docs/docs/install.md:157]:
```suggestion
NetKet v3 API is centered around [flax](https://flax.readthedocs.io), a JAX-based library providing components to define and use neural network models.
```

File: docs/docs/hilbert.md [hunk 8]
```diff
@@ -113,17 +113,17 @@ This information can be extracted with the {meth}`~netket.hilbert.DiscreteHilber
...
-You can also obtain the total size of the hilbert space by invoking {attr}`~netket.hilbert.DiscreteHilbert.n_states`, which in general is equivalent to calling `np.prod(hi.shape)`.
+You can also obtain the total size of the hilbert space by invoking {attr}`~DiscreteHilbert.n_states`, which in general is equivalent to calling `np.prod(hi.shape)`.
 
 ```python
 >>> hi.n_states
 96
 ```
 
-Do bear in mind that this attribute only works if the hilbert space is indexable ({attr}`~netket.hilbert.DiscreteHilbert.is_indexable`), which is True when it has a size smaller than $ 2^{64} $. 
+Do bear in mind that this attribute only works if the hilbert space is indexable ({attr}`~DiscreteHilbert.is_indexable`), which is True when it has a size smaller than $ 2^{64} $.
```
Comment (femtobit) [inline docs/docs/hilbert.md:123]:
```suggestion
Bear in mind that this attribute only works if the Hilbert space is indexable ({attr}`~DiscreteHilbert.is_indexable`), which is true if it has a dimension smaller than $2^{64}$. 
```
`Hilbert.size` is the number of sites in NetKet, so `dimension` is probably the better term here, right?

File: docs/docs/hilbert.md [hunk 9]
```diff
@@ -83,10 +83,10 @@ Please do open an issue or a feature request on the GitHub repository if you enc
 
 ## The `DiscreteHilbert` interface
 
-{ref}`netket.hilbert.DiscreteHilbert` is also an abstract class from which any hilbert space with countable (or discrete) local degrees of freedom must inherit.
+{class}`DiscreteHilbert` is also an abstract class from which any hilbert space with countable (or discrete) local degrees of freedom must inherit.
 Examples of such spaces are spins or bosons on a lattice.
 
-You can always probe their {attr}`~netket.hilbert.DiscreteHilbert.shape`, which returns a tuple 
+You can always probe their {attr}`~DiscreteHilbert.shape`, which returns a tuple 
 with the size of the hilbert space on every site/degree of freedom.
```
Comment (femtobit) [inline docs/docs/hilbert.md:90]:
```suggestion
with the size of the Hilbert space on every site/degree of freedom.
```

File: netket/utils/group/_group.py (hunk 1 above)
Comment (PhilipVinc) [reply netket/utils/group/_group.py:115]: why? Maybe I was missing a comma but it should read a bit like

Assuming the definitions .... then, A is equivalent to B.

Comment (PhilipVinc) [reply netket/utils/group/_group.py:115]: hmm. ok, grammatically you are right..

**Comment by PhilipVinc** [comment]
ok, fixed and rebated.

File: docs/api/nn.md [hunk 10]
```diff
@@ -6,7 +6,7 @@
 
 ```
 
-This sub-module wraps and re-exports [flax.nn](https://flax.readthedocs.io/en/latest/flax.linen.html).
+This sub-module extends [flax.linen](https://flax.readthedocs.io/en/latest/flax.linen.html) with layers and tools that are useeful to applications in quantum physics.
```
Comment (femtobit) [inline docs/api/nn.md:9]:
```suggestion
This sub-module extends [flax.linen](https://flax.readthedocs.io/en/latest/flax.linen.html) with layers and tools that are useful to applications in quantum physics.
```

**Review by femtobit — APPROVED** [review]
Sorry, one more typo. :sweat_smile: Otherwise looks good to me.

**Pull request merged**